lib/diaspora_federation/properties_dsl.rb

```diff
@@ -71,11 +71,18 @@ def class_prop_names
     def define_property(name, type, opts={})
       raise InvalidName unless name_valid?(name)
 
-      xml_name = name
       if opts.has_key? :xml_name
         raise ArgumentError, "xml_name is not supported for nested entities" unless type == String
         xml_name = opts[:xml_name]
         raise InvalidName, "invalid xml_name" unless name_valid?(xml_name)
+      elsif type == String
+        xml_name = name
+      elsif type.instance_of?(Array)
+        xml_name = type.first.entity_name.to_sym
+      elsif type.ancestors.include?(Entity)
+        xml_name = type.entity_name.to_sym
+      else
+        raise ArgumentError, "unknown type #{type} supplied"
```

Comment:
Well, that's kinda philosophical. I personally prefer to keep public API surface as low as possible, allowing for easier and faster iteration without breaking other parts. Hence I prefer private instance methods if they're used only within the class to extract redundant logic, even if it doesn't depend on instance data.

The other viewpoint is to focus on re-usability, the argument is that just because I don't have another usecase at hand *at the moment*, that doesn't mean me or somebody else won't in the future, thus making it public is beneficial to that aspect. The extreme version of that is to make no private methods at all.

I can't say there's a clear consensus in the (Ruby) community for either variant, but mixing both approaches usually leads to the downsides of both without the benefits of either. So at least within a gem/project I would pick and stick to one. Which one to pick is atm @SuperTux88 decision.